src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1674,121 +1728,161 @@ Out valueGenerator(const Node& node,
-     }},
-    {"^ActorId",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<ActorIdIntGenerator>(node, generatorArgs);
-     }},
-    {"^RandomInt", int64GeneratorBasedOnDistribution},
-    {"^RandomDouble", doubleGeneratorBasedOnDistribution},
-    {"^Now",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<NowGenerator>(node, generatorArgs);
-     }},
-    {"^RandomDate",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<RandomDateGenerator>(node, generatorArgs);
-     }},
-    {"^Date",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<DateGenerator>(node, generatorArgs);
-     }},
-    {"^ObjectId",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<ObjectIdGenerator>(node, generatorArgs);
-     }},
-    {"^Verbatim",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return valueGenerator<true, UniqueAppendable>(node, generatorArgs, allParsers);
-     }},
-    {"^Inc",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<IncGenerator>(node, generatorArgs);
-     }},
-    {"^TwoDWalk",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<TwoDWalkGenerator>(node, generatorArgs);
-     }},
-    {"^IncDate",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<IncDateGenerator>(node, generatorArgs);
-     }},
-    {"^Array",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<ArrayGenerator>(node, generatorArgs, allParsers);
-     }},
-    {"^Object",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<ObjectGenerator>(node, generatorArgs, allParsers);
-     }},
-    {"^Cycle",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<CycleGenerator>(node, generatorArgs, allParsers);
-     }},
-    {"^Repeat",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<RepeatGenerator>(node, generatorArgs, allParsers);
-     }},
-    {"^FixedGeneratedValue",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<CycleGenerator>(node, generatorArgs, allParsers, 1);
-     }},
-    {"^BinData",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         return std::make_unique<BinDataGenerator>(node, generatorArgs);
-     }},
-    {"^BinDataSensitive",
-     [](const Node& node, GeneratorArgs generatorArgs) {
-         // TODO: PERF-4467 Update this to bsoncxx::binary_sub_type::sensitive.
-         return std::make_unique<BinDataGenerator>(
-            node, generatorArgs, bsoncxx::binary_sub_type(0x8));
-     }},
-};
+const auto [allParsers, arrayParsers, dateParsers, doubleParsers, intParsers, stringParsers] = [] () {
```

Comment:
What do we gain by the changes in this block?

Reply:
The main benefit is a deduplication of parser entries. It's really easy to only add the parser to just `arrayParsers`, for example, and forget to add it to `allParsers`. This makes the `allParsers` map inherit mappings from existing parser maps, to avoid the risk of accidentally forgetting to map a parser.

Reply:
That makes a lot of sense, and i know that error mode has occurred. Could you add a comment explaining this, and include instructions for what to do when adding a new parser--I think that would now be just add to one list, and it's then automatically included in the others appropriately.

Reply:
I spent some time debugging this yesterday, too, so I figured it'd be a good change.